### PR TITLE
enhancement: improve search feature

### LINKED ISSUE #39: Entering a lone parenthesis in the search leads to internal server error
The search mechanism  uses `re` to compile the string we give it, but this method seems to have a problem if the string we pass it contains a lone parenthesis. 

See Sentry RERO-ILS-1RK

### PR DESCRIPTION
**Pull request by PascalRepond**
* Closes https://github.com/rero/flask-wiki/issues/39
* Escapes query before searching
* Search results are now sorted by number of occurences
of the search term
* Cleans up api.py code

Co-Authored-by: Pascal Repond <pascal.repond@rero.ch>

### CODE REVIEW

**Review by BadrAly — APPROVED** [review]

File: flask_wiki/api.py [hunk 1]
```diff
@@ -399,24 +389,37 @@ def current_language(self):
     def languages(self):
         return current_app.config.get('WIKI_LANGUAGES')
 
-    def search(self, term, ignore_case=True, attrs=['title', 'tags', 'body']):
+    def search(self, term, ignore_case=True, attrs=None):
+        if attrs is None:
+            attrs = ['title', 'tags', 'body']
         pages = self.index()
 
+        for page in pages:
```
Comment (zannkukai) [inline flask_wiki/api.py:397]: could be removed, see next comment

File: flask_wiki/api.py [hunk 2]
```diff
@@ -399,24 +389,37 @@ def current_language(self):
...
+
         matched = []
         for page in current_language_pages:
             for attr in attrs:
-                if regex.search(getattr(page, attr)):
-                    matched.append(page)
-                    break
-        return matched
+                if found := re.findall(regex, getattr(page, attr)):
+                    page["score"] += len(found)
```
Comment (zannkukai) [inline flask_wiki/api.py:418]: `page['score'] = page.setdefault('score', 0) + len(found)`

**Review by zannkukai — APPROVED** [review]
comestic comments

File: flask_wiki/api.py (hunk 2 above)
Comment (PascalRepond) [reply flask_wiki/api.py:418]: I just tried this and it prompts an error because `page` is a `Page` object and not a dictionnary.

**Pull request merged**